pubsublite/producer.go

```diff
@@ -202,7 +190,11 @@ func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 		topic := p.cfg.TopicRouter(event)
 		producer, ok := p.producer[topic]
 		if !ok {
-			return fmt.Errorf("pubsublite: unable to find producer for %s", topic)
+			publisher, err := newPublisher(ctx, p.cfg, topic)
+			if err != nil {
+				return fmt.Errorf("pubsublite: failed creating publisher: %w", err)
+			}
+			p.producer[topic] = publisher
```

Comment:
This method acquires a read lock, yet it mutates the producer map, which will lead to races when this method is called concurrently.

Looking at the implementation and how we use the map, I think it may be best to use a `sync.Map`, so we don't have to introduce a new Mutex.

Reply:
Good catch! Updated :+1: